### ISSUE TITLE
IMPORTANT July 2021: Chrome private network access restriction with hot module reloading

### ISSUE THREAD

**Issue by FlavioSB** [JeffreyWay/laravel-mix#3033]
-   Laravel Mix Version: 6.0.24
-   Node Version: 14.16.0
-   NPM Version: 6.14.11
-   OS: Windows 10 Pro

### Description:
Using HMR spits out a deprecation warning from Google Chrome ([Google page](https://chromestatus.com/feature/5436853517811712) | [Google developer blog](https://developer.chrome.com/blog/private-network-access-update/) ), which sounds like it will ultimately break the functionality to request data from one site (in my example a homestead site "site.test") to a more private site (in this case the websocket). This picture helped me understand it, I think my situation is the private network -> local device example: 
![nSGfU9xMJxIy6lMjJiMx](https://user-images.githubusercontent.com/80675031/125247485-4b1c2c80-e2f3-11eb-8fcd-d20c01d0e204.png)

**Deprecation warning:**
```[Deprecation] The website requested a subresource from a network that it could only access because of its users' privileged network position. These requests expose non-public devices and servers to the internet, increasing the risk of a cross-site request forgery (CSRF) attack, and/or information leakage. To mitigate these risks, Chrome deprecates requests to non-public subresources when initiated from non-secure contexts, and will start blocking them in Chrome 92 (July 2021). See https://chromestatus.com/feature/5436853517811712 for more details.```

### Steps To Reproduce:
Have a private network setup like mysite.test and start hmr with the `mix watch --hot` command. There has to be some kind of request to a subresource, in my case its just the request to any page.

**Comment by FlavioSB** [comment]
I feel like this is a VERY important matter. If this really starts blocking these requests a lot of people can't keep developing their apps with HMR (which is a really neat and popular feature). Switching to `watch` does work but isnt a solution at all.

**Comment by thecrypticace** [comment]
Sorry, I've been incredibly busy. I investigated this already back in May and you can see my response here: https://github.com/JeffreyWay/laravel-mix/issues/2987#issuecomment-847100684 Basically this is not something mix should be doing but is on webpack-dev-server to change if it hasn't already done so.

**Comment by FlavioSB** [comment]
No problem & thank you for answering. 

That the problem needs to be fixed on webpack-dev-servers side makes sense. Mistake on my end, I just felt it was a very important matter, not only to myself but a lot of other people.

**Comment by FlavioSB** [comment]
I made an issue post on their GitHub page and referenced this one.

So this issue is on hold until they fixed it, so laravel-mix can update the package.

**Comment by FlavioSB** [comment]
webpack/webpack-dev-server#3585

Apparently it's not on their part either, I have to proxy all requests to the same network to prevent this issue. A lot of work for nothing though...

**Comment by thecrypticace** [comment]
Yeah proxying isn't a solution. It _is_ on them technically to add the headers to override this but I'll see if there's something we can do. I'm just not sure how much if anything.

It almost certainly won't be this week though but I'll keep this in my head and maybe try something over the weekend.

**Comment by FlavioSB** [comment]
I'm afraid you're better off discussing this with them yourselves, I can't be the messenger for something I don't fully understand yet. I feel you both understand this matter better than I do.
Thank you for looking into it!

**Comment by thecrypticace** [comment]
Oh yeah totally understand. I don't expect you too at all. Honestly it's complicated for me to understand too but I think I've got somewhat of a grasp on what we may be able to do to workaround the issues — hopefully.
